### PR TITLE
Stripe tests

### PR DESCRIPTION
**Pull request by lemieux**
I added tests for the action component and the stripe service.

@Mario54 @mspensieri @alavers

### CODE REVIEW

File: test/specs/components/action.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+        sandbox.stub(userService, 'immediateUpdate');
+        userService.immediateUpdate.resolves();
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    after(() => {
+        Object.defineProperty(AppStore, 'store', {
```
Comment (alavers) [inline test/specs/components/action.spec.jsx]: Why is this necessary?

File: test/specs/components/action.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+            text: 'action text',
+            uri: 'action uri'
+        };
+
+        beforeEach(() => {
+            component = TestUtils.renderIntoDocument(<ActionComponent {...props} />);
+            componentNode = ReactDOM.findDOMNode(component);
+        });
+
+        it('should render a link with target blank', () => {
```
Comment (alavers) [inline test/specs/components/action.spec.jsx]: `beforeEach` and `it` could be merged. Do you plan to add more `it`s here?

File: test/specs/components/action.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+            link.target = '_self';
+        });
+    });
+
+    describe('buy action with stripe keys and offered state', () => {
+        const props = {
+            type: 'buy',
+            text: 'action text',
+            uri: 'fallback uri',
+            currency: 'USD',
```
Comment (alavers) [inline test/specs/components/action.spec.jsx]: nit, stripe lower cases currency and we should probably follow suit

Comment (lemieux) [inline test/specs/components/action.spec.jsx]: react-stripe-checkout expects it to be uppercased.

File: test/specs/components/action.spec.jsx (hunk 2 above)
Comment (lemieux) [inline test/specs/components/action.spec.jsx]: probably not, but I've been doing that in the whole test suite..

File: test/specs/components/action.spec.jsx (hunk 1 above)
Comment (lemieux) [inline test/specs/components/action.spec.jsx]: since `store` is not a method of `AppStore` it cannot be stubbed. The mock store above takes care of replacing the value by a mock. This part restores it at the initial value.

I really need to try babel-rewire again.

File: test/specs/components/action.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+            state: 'offered'
+        };
+
+        beforeEach(() => {
+            mockStore(sandbox, {
+                app: {
```
Comment (alavers) [inline test/specs/components/action.spec.jsx]: Lots of repeated state here. Tests might be more terse if you returned this state via a `getDefaultState()` helper, and then had each test simply modify the props you want for each scenario.

Lots of code reuse opportunities in general

File: test/specs/components/action.spec.jsx (hunk 3 above)
Comment (lemieux) [inline test/specs/components/action.spec.jsx]: in fact it doesn't really matter, I do `this.props.currency.toUpperCase()` before passing it

File: test/specs/components/action.spec.jsx (hunk 4 above)
Comment (lemieux) [inline test/specs/components/action.spec.jsx]: true I'll simplify this

File: test/specs/services/stripe-service.spec.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import sinon from 'sinon';
+import { getMock } from 'test/mocks/core';
```
Comment (alavers) [inline test/specs/services/stripe-service.spec.js]: nit, should maybe call this`createMock`

File: test/specs/services/stripe-service.spec.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import { getMockedStore } from 'test/utils/redux';
+import * as coreService from 'services/core';
+import { createTransaction, getAccount } from 'services/stripe-service';
+
+const AppStore = require('stores/app-store');
+
+const store = AppStore.store;
+
+function mockStore(s, state = {}) {
```
Comment (alavers) [inline test/specs/services/stripe-service.spec.js]: Maybe this method should be moved into utils?

File: test/specs/services/stripe-service.spec.js (hunk 5 above)
Comment (lemieux) [inline test/specs/services/stripe-service.spec.js]: true it's a mock factory

File: test/specs/services/stripe-service.spec.js (hunk 6 above)
Comment (lemieux) [inline test/specs/services/stripe-service.spec.js]: yes

File: test/specs/components/action.spec.jsx (hunk 1 above)
Comment (Mario54) [inline test/specs/components/action.spec.jsx]: For now, what about adding a `restore` method on the mockedStore returned from the `mockStore` function?

e.g.

```javascript
function mockStore(storeBeingMocked, s, state = {}) {
     var originalStore = storeBeingMocked.store;
     var mockedStore = getMockedStore(s, state);
 
     Object.defineProperty(storeBeingMocked, 'store', {
         get: () => {
             return mockedStore;
         }
     });

     mockedStore.restore = function() {
        Object.defineProperty(storeBeingMocked, 'store', {
            get: () =>
                return originalStore;
           }
     });
 
     return mockedStore;
 }
```

Not sure if that would work or if it's a good idea

Comment (lemieux) [inline test/specs/components/action.spec.jsx]: I'll do something along these lines yes

File: test/specs/components/action.spec.jsx (hunk 3 above)
Comment (lemieux) [inline test/specs/components/action.spec.jsx]: fixed

File: test/specs/components/action.spec.jsx (hunk 1 above)
Comment (lemieux) [inline test/specs/components/action.spec.jsx]: fixed

File: test/specs/services/stripe-service.spec.js (hunk 5 above)
Comment (lemieux) [inline test/specs/services/stripe-service.spec.js]: fixed

File: test/specs/services/stripe-service.spec.js (hunk 6 above)
Comment (lemieux) [inline test/specs/services/stripe-service.spec.js]: done

File: test/specs/components/action.spec.jsx (hunk 4 above)
Comment (lemieux) [inline test/specs/components/action.spec.jsx]: done

**Comment by lemieux** [comment]
@alavers nit fixed

**Comment by Mario54** [comment]
:+1:

**Comment by alavers** [comment]
:+1:
